### PR TITLE
authentication module

### LINKED ISSUE #47: [BE] Implement Member Sign In API
## Business Requirements
As a visitor, I want to be able to sign in as a member to the platform so that I can use it.

## Acceptance Criteria
- [ ] 1. There should be an endpoint to sign in members by email address and password
- [ ] 2. There should be request validation for each field.
- [ ] 3. There should be a API rate limiting of 5 requests per minute.
- [ ] 4. Only members should be able to sign in via this endpoint, any other role should get an error message.
- [ ] 5. There should be API documentation generated by Scribe so that the frontend team can easily integrate it.

### LINKED ISSUE #107: [BE] Re-work the sign in endpoints to use the Laravel Sanctum single page application approach
### Task description

We currently have a few sign in endpoints which use Laravel Sanctum but it uses the mobile authentication approach with the plain text token being returned on the first request.

We need to use the Single Page Application approach with the CSRF token since we have a single page application that needs to integrate these endpoints.

More information can be found in the [Laravel documentation](https://laravel.com/docs/11.x/sanctum#spa-authentication) on how to do this.

The redirect paths should be the following:
/content/dashboard - content
/admin/dashboard - admin
/dashboard - platform

### LINKED ISSUE #114: [BE] Implement the Authentication module
### Task Description

Currently we have a very similar authentication-related logic in multiple modules.

Please create a new module called Authentication and move all of the logic in it but keep in mind that:
- Only members can sign in to the platform
- Only administrators can sign in to the administration panel
- Only content managers and administrators can sign in to the content panel

### Related tasks
- #107
- #47

### PR DESCRIPTION
**Pull request by calexandar**
#114

### CODE REVIEW

File: app/Authentication/database/migrations/2024_07_09_083223_set_up_authentication_module.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
```
Comment (davorminchorov) [inline app/Authentication/database/migrations/2024_07_09_083223_set_up_authentication_module.php]: You can remove this file

File: app/Authentication/src/Http/Controllers/AdminAuthController.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+use App\Authentication\Http\Requests\AdminLogoutRequest;
+
+
+class AdminAuthController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('auth:sanctum')->only(['index', 'logout']);
+        $this->middleware('treblle');
+        $this->middleware('stateful');
```
Comment (davorminchorov) [inline app/Authentication/src/Http/Controllers/AdminAuthController.php]: You could apply these middleware statements to the routes directly to keep it consistent.

This applies to all controllers.

File: app/Authentication/src/Http/Controllers/ContentManagerAuthController.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    public function logout(ContentManagerLogoutRequest $request)
+    {
+
+        $request->session()->invalidate();
+
+        $request->session()->regenerateToken();
+
+        return response()->json([
+            'message' => 'Logged Out!'
```
Comment (davorminchorov) [inline app/Authentication/src/Http/Controllers/ContentManagerAuthController.php]: All messages should be in macedonian.

File: app/Authentication/tests/AuthenticationServiceProviderTest.php [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?php
```
Comment (davorminchorov) [inline app/Authentication/tests/AuthenticationServiceProviderTest.php]: You can remove this file.

File: app/Content/routes/content-routes.php [hunk 5]
```diff
@@ -7,10 +7,10 @@
 use Illuminate\Support\Facades\Route;
 
 Route::group(['prefix' => '/content', 'middleware' => ['treblle']], function () {
-    Route::post('/login', [AuthController::class, 'login'])->middleware(['throttle:login']);
+    //Route::post('/login', [AuthController::class, 'login'])->middleware(['throttle:login']);
 
     Route::middleware(['auth:sanctum', 'verified'])->group(function () {
-        Route::post('/logout', [AuthController::class, 'logout']);
+        //Route::post('/logout', [AuthController::class, 'logout']);
```
Comment (davorminchorov) [inline app/Content/routes/content-routes.php]: You can remove the unused / commented out code everywhere.

**Review by davorminchorov — CHANGES_REQUESTED** [review]
It looks like you forgot to add the attributes for the API docs. We need to group stuff by module and make sure that the URL points to staging for testing purposes.

Check how we do it in other parts of the app and check the Laravel Scribe docs for more info.

File: .scribe/intro.md [hunk 6]
```diff
@@ -3,7 +3,7 @@
 The documentation
 
 <aside>
-    <strong>Base URL</strong>: <code>https://staging-api.learnhub.mk</code>
```
Comment (davorminchorov) [inline .scribe/intro.md]: You have to add the APP_STAGING_URL .env variable and assign it to this URL in order for the docs to generate with the correct URL and the "try it out" feature to work.

Comment (calexandar) [reply .scribe/intro.md]: I added APP_STAGING_URL but it still fails it requires 

# social accounts
GITHUB_CLIENT_ID=
GITHUB_CLIENT_SECRET=
GITHUB_REDIRECT_URI=

GOOGLE_CLIENT_ID=
GOOGLE_CLIENT_SECRET=
GOOGLE_REDIRECT_URI=

LINKEDIN_CLIENT_ID=
LINKEDIN_CLIENT_SECRET=
LINKEDIN_REDIRECT_URI=

Comment (davorminchorov) [reply .scribe/intro.md]: Fails on generation? Add some random values and see if it will fix it.

**Comment by davorminchorov** [comment]
Don't forget to fix the code styles by running `./vendor/bin/pint` at the end so that the coding style is consistent.

File: .scribe/intro.md (hunk 6 above)
Comment (calexandar) [reply .scribe/intro.md]: It doesn't work with dummy text. I will need to create accounts for socillite account and then create scribe docs for this section. I added update scribe docs with new values for other routes and header type. There were some bugs I fixed them for scribe:generate

**Review by davorminchorov — APPROVED** [review]

**Pull request merged**